### ISSUE TITLE
Astro frontend should redirect users to login page when authentication has failed

### ISSUE THREAD

**Issue by cfc7-unikent** [unikent/astro#180]
Currently, I come in after leaving Astro open overnight and my session with the backend has lapsed. 

However, the frontend lets me carry on working until it becomes clear to me that my authentication is failing and I manually log out and log back in. 

We should listen for auth failures in the front end (via a http interceptor in the api?) and when we encounter one redirect the user to the login page.

**Comment by theninja** [comment]
This is something that's also bothered me for a while. This is due to the API tokens being long lived, so they are still valid, even though the rest of the session has expired. The front-end will keep sending a valid API token until you refresh the page (the config class then no longer has access to the API token, as it's only given to the client after successful sign in). There's no way of listening for this as it happens before we have access to the API via SSO.

Basically we need to rethink authentication, using a more modern approach, possibly JWT (a short lived token + a refresh token). This is linked to #47.

**Comment by sy238-unikent** [comment]
see also issue #47
